### ISSUE TITLE
[Q] Error trace with Sweeps

### ISSUE THREAD

**Issue by GillesJ** [wandb/client#1971]
I was wondering were we can obtain the error trace of the internal model with hyperoptim sweeps.

Right now all I get in logs and stdout is:
```
Run axtblwew errored: TypeError("'float' object is not subscriptable")
wandb: ERROR Run axtblwew errored: TypeError("'float' object is not subscriptable")
```

A line number or ideally a error backtrace would go a long way towards debugging complex models/configs.

**Comment by ariG23498** [comment]
Hey @GillesJ 
Could you share your `sweep.yaml` file with us?
We will keep the better error loggin in mind, thanks for the ticket! 😄

**Comment by GillesJ** [comment]
I programmatically set the hyperparam config in the Python script as per [this tutorial](https://towardsdatascience.com/hyperparameter-optimization-for-optimum-transformer-models-b95a32b70949) with layer-wise nested parameter settings:

```
layer_parameters = {f"layer_{i}-{i + 6}": {"min": 0.0, "max": 5e-5} for i in range(0, 24, 6)}

sweep_config = {
    "name": "layerwise-sweep-batch-16",
    "method": "bayes",
    "metric": {"name": "f1_macro", "goal": "maximize"},
    "parameters": {
        "num_train_epochs": {"min": 1, "max": 40},
        "params_classifier.dense.weight": {"min": 0, "max": 1e-3},
        "params_classifier.dense.bias": {"min": 0, "max": 1e-3},
        "params_classifier.out_proj.weight": {"min": 0, "max": 1e-3},
        "params_classifier.out_proj.bias": {"min": 0, "max": 1e-3},
        **layer_parameters,
    },
    "early_terminate": {"type": "hyperband", "min_iter": 6,},
}

sweep_id = wandb.sweep(sweep_config, project="Hyperparameter Optimization")
```

**Comment by ariG23498** [comment]
>"params_classifier.dense.weight": {"min": 0, "max": 1e-3},
        "params_classifier.dense.bias": {"min": 0, "max": 1e-3},
        "params_classifier.out_proj.weight": {"min": 0, "max": 1e-3},
        "params_classifier.out_proj.bias": {"min": 0, "max": 1e-3},

My guesses are the keys with the `.` are not well handled as parameter names. We automatically flatten nested parameters, so if you pass us a dictionary we will turn it into a dot-separated name. If we have dot-seperated names there is a clash of conflicts.

Could you check to change these keys and report back, please?

**Comment by GillesJ** [comment]
Changed the names to not include '.' but got the same error. 

I have confirmed that the train() methods works just fine though, so the error is somewhere in the wandb sweep handling.

`pdb` and `pycharm` interactive debuggers don't seem to work when using `wand.agent` so it's difficult to figure out what's going on.

**Comment by ariG23498** [comment]
Could you share a colab for us to reproduce this error?

**Comment by GillesJ** [comment]
Thanks for your responses. I don't have colab notebooks set up and I currently can not spare the time. Code used is from [this tutorial](https://towardsdatascience.com/hyperparameter-optimization-for-optimum-transformer-models-b95a32b70949).

**Comment by NaRuecker** [comment]
@GillesJ : I'm having exactly the same issue, after following the same tutorial. Could you figure out what was going wrong?
@ariG23498: I don't have any dots in the parameters... please, see below.

sweep_config = {
    "name": "trial 1",
    "method": "bayes",
    "metric": {"name": "accuracy", "goal": "maximize"},
    "parameters": {
        "num_train_epochs": {"min": 3, "max": 20},
        "learning_rate": {"min":0, "max": 0.1},
    },
    "early_terminate": {"type": "hyperband", "min_iter": 6,}
}

Many thanks to both of you for your time!

**Comment by dannygoldstein** [comment]
hi @NaRuecker,

I believe this is due to your mixing of float and int types in `learning_rate`,

Please try changing 

`{"min":0, "max": 0.1}` -> `{"min":0.0, "max": 0.1}`

Does that fix the issue?

**Comment by NaRuecker** [comment]
@dannygoldstein : Great catch, but unfortunately same error.

**Comment by GillesJ** [comment]
Working on integrating `wandb` into another codebase now and the lack of default python stack trace makes development really difficult. The original question remains unanswered as focus shifted towards specific issues, but I still want to know:

**How can I get logging or wandb to print/log the default interpreter stack trace on error**, especially with context/line number for improved debugging?
